File: backend/app/schemas/ai.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class AiConfig(BaseModel):
    enabled: bool = False
    api_base_url: str | None = None
    api_key: str | None = None
    model: str | None = None
    timeout_seconds: int = 20


class AiConfigRead(BaseModel):
    enabled: bool = False
    api_base_url: str | None = None
    api_key_set: bool = False
    model: str | None = None
    timeout_seconds: int = 20
    last_test_result: str | None = None
    last_test_at: datetime | None = None
    last_call_at: datetime | None = None


class AiInsightRequest(BaseModel):
    batch_id: int | None = None
    calculation_profile_id: int | None = None
    baseline_plan_id: int | None = None
    building: str | None = None
    mode: str = Field(default="dashboard_summary")


class AiInsightResponse(BaseModel):
    enabled: bool
    generated_text: str
    fallback_text: str
    source: str
    error_message: str | None = None


class AiConnectionTestRequest(BaseModel):
    api_base_url: str | None = None
    api_key: str | None = None
    model: str | None = None
    timeout_seconds: int = 20


class AiConnectionTestResponse(BaseModel):
    success: bool
    message: str
    tested_at: datetime | None = None


class AiPromptTemplateBase(BaseModel):
    name: str
    code: str
    description: str | None = None
    prompt_template: str
    is_active: bool = True


class AiPromptTemplateCreate(AiPromptTemplateBase):
    project_id: int | None = None


class AiPromptTemplateUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    prompt_template: str | None = None
    is_active: bool | None = None


class AiPromptTemplateRead(AiPromptTemplateBase):
    id: int
    project_id: int | None = None
    is_builtin: bool = False
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class AiCallLogRead(BaseModel):
    id: int
    project_id: int | None = None
    batch_id: int | None = None
    mode: str
    model: str | None = None
    source: str
    success: bool
    error_message: str | None = None
    prompt_tokens: int | None = None
    completion_tokens: int | None = None
    input_summary_length: int | None = None
    output_length: int | None = None
    duration_ms: int
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/ai.ts
export type AiConfigRead = {
  enabled: boolean
  api_base_url?: string | null
  api_key_set: boolean
  model?: string | null
  timeout_seconds: number
  last_test_result?: string | null
  last_test_at?: string | null
  last_call_at?: string | null
}

export type AiConfigPayload = {
  enabled: boolean
  api_base_url?: string | null
  api_key?: string | null
  model?: string | null
  timeout_seconds: number
}

export type AiInsightResponse = {
  enabled: boolean
  generated_text: string
  fallback_text: string
  source: 'ai' | 'rule_fallback'
  error_message?: string | null
}

export type AiPromptTemplate = {
  id: number
  project_id?: number | null
  name: string
  code: string
  description?: string | null
  prompt_template: string
  is_builtin: boolean
  is_active: boolean
  created_at: string
  updated_at: string
}

export type AiCallLog = {
  id: number
  project_id?: number | null
  batch_id?: number | null
  mode: string
  model?: string | null
  source: 'ai' | 'rule_fallback'
  success: boolean
  error_message?: string | null
  prompt_tokens?: number | null
  completion_tokens?: number | null
  input_summary_length?: number | null
  output_length?: number | null
  duration_ms: number
  created_at: string
}
